src/early-late-bound.md

```diff
@@ -0,0 +1,90 @@
+# Early and Late Bound Variables
+
+In Rust, generic types are generally [_universally_ quantified][quant].
```

Comment:
well, I think I would phrase this like so

In Rust, item definitions (like `fn`) can often have generic parameters, which are always [_universally_ quantified][quant]. That is, if you have a function like

```rust
fn foo<T>(x: T) { }
``` 

this function is defined "for all T" (not "for some specific T", which would be existentially quantified).

While Rust *items* can be quantified over types, lifetimes, and constants, the types of values in Rust are only ever quantified over lifetimes. So you can have a type like `for<'a> fn(&'a u32)`, which represents a function pointer that takes a reference with any lifetime, or `for<'a> dyn Trait<'a>`, which is a `dyn` trait for a trait implemented for any lifetime; but we have no type like `for<T> fn(T)`, which would be a function that takes a value of *any type* as a parameter. This is a consequence of monomorphization -- to support a value of type `for<T> fn(T)`, we would need a single function pointer that can be used for a parameter of any type, but in Rust we generate customized code for each parameter type. 

One consequence of this asymmetry is a weird split in how we represesent some generic types: _early-_ and _late-_ bound parameters.

Reply:
I replaced the intro with this (let me know if I didn't replace enough). Sorry, I'm only slowly starting to understand this distinction; you're intro helps a lot.